src/components/SchedulePanel/components/EditDescriptionDialog/index.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogActions from '@material-ui/core/DialogActions';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import { gordonColors } from '../../../../theme';
+import './editdescriptiondialog.css';
+
+export default class EditDescriptionDialog extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      descInput: '',
+      formErrors: {
+        descInput: '',
+      },
+      editDescriptionOpen: false,
+    };
+    this.handleEditDescriptionClose = this.handleEditDescriptionClose.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.descInput !== nextProps.descriptiontext) {
+      this.setState({ descInput: nextProps.descriptiontext });
+    }
+  }
+
+  handleSubmit = e => {
+    e.preventDefault();
+    var desc = this.state.descInput;
+    this.props.onDialogSubmit(desc);
+    this.handleClose();
+  };
+
+  handleChange = name => e => {
+    this.setState({ [name]: e.target.value }, () => {});
+  };
+
+  handleEditDescriptionClose = () => {
+    this.setState({ editDescriptionOpen: false });
+  };
+
+  handleClose = () => {
+    this.props.handleEditDescriptionClose();
```

Comment:
Do you want the one you declared in this component, or the one from props?

Reply:
Change made.